module/userExtension/helper.js

```diff
@@ -800,6 +807,163 @@ module.exports = class UserExtensionHelper {
+                        entity++
+                    ) {
+
+                        let singleEntity = entityDocuments[entity];
+
+                        let entityObj = {
+                            name : singleEntity.metaInformation.name,
+                            externalId : singleEntity.metaInformation.externalId,
+                            entityType : singleEntity.entityType,
+                            entityTypeId : singleEntity.entityTypeId,
+                            _id : singleEntity._id
+                        };
+
+                        let telementryObj = {
+                            [`${entityObj.entityType}_name`] : entityObj.name,
+                            [`${entityObj.entityType}_id`] : entityObj._id,
+                            [`${entityObj.entityType}_externalId`] : entityObj.externalId
+                        };
+
+                        let relatedEntities = 
+                        await entitiesHelper.relatedEntities(
+                            entityObj._id,
+                            entityObj.entityTypeId,
+                            entityObj.entityType,
+                            [
+                                "metaInformation.externalId",
+                                "metaInformation.name",
+                                "entityType",
+                                "entityTypeId",
+                                "_id"
+                            ]
+                        );
+
+                        if( relatedEntities.length > 0 ) {
+                            entityObj["relatedEntities"] = 
+                            relatedEntities.map(
+                                entity => {
+                                    telementryObj[`${entity.entityType}_name`] = 
+                                    entity.metaInformation.name;
+                
+                                    telementryObj[`${entity.entityType}_id`] = 
+                                    entity._id;
+                
+                                    telementryObj[`${entity.entityType}_externalId`] = 
+                                    entity.metaInformation.externalId;
+                
+                                    return {
+                                      name : entity.metaInformation.name,
+                                      externalId : entity.metaInformation.externalId,
+                                      entityType : entity.entityType,
+                                      entityTypeId : entity.entityTypeId,
+                                      _id : entity._id
+                                    }
+                                }
+                            ) 
+                        }
+
+                        telemetry.push(telementryObj);
+                        entities.push(entityObj)
+                    }
+                }
+            }
+  
+            let user = 
+            await elasticSearchData.get(userInformation);
```

Comment:
@aman-tunerlabs - No need to check and get and update, just do something like upsert and override the existing document if it already exists.